bluesky/simulators.py

```diff
@@ -114,3 +114,276 @@ def check_limits(plan):
+        """
+        Returns the combination est_time/std_dev pairs et_1 and et_2.
+        This function returns the combination of est_time_1 and est_time_2, combined using the 
+        method definedby 'method'.
+
+        Parameters
+        ----------
+        est_time_1, est_time_2 : tuples.
+            The tuples containing the est_time/std_dev tuples to be combined.
+        method : string, optional.
+            The method to use for the combination of est_time_1 and est_time_2, default is 
+            'sum'.
+
+        Return Parameters
+        -----------------
+        out_est_time : list.
+            The combined est_time/std_dev tuple.
+        """
+        out_est_time = [0,0]
+        if method == 'sum':
+            out_est_time[0] = est_time_1[0] + est_time_2[0]
+            out_est_time[1] = est_time_1[1] + est_time_2[1]
+
+        elif method == 'max':
+            if est_time_1[0] < est_time_2[0]:
+                out_est_time = est_time_2 
+
+            else:
+                out_est_time = est_time_1
+
+        return out_est_time
+
+
+
+    def obj_est_time(msg, val_dict):
+        """
+        Returns the est_time/std_dev pair for the object referenced in msg.
+        This function returns the est_time/std_dev pair for the object referenced in msg.obj and 
+        for the command type reference in msg.command.
+
+        Parameters
+        ----------
+        msg : message.
+            The first message of the group that was detected in plan_ETA.
+        val_dict : dict.
+            A dictionary containing information on values updated during the plan. It has key:arg 
+            pairs with keys relating to message components where each arg is a dictionary 
+            containing devices (detectors, axes) that have been updated, as keywords and a value 
+            indicating the update status. For 'set' the update status is the latest 'position' that
+            it is to be set and for 'trigger' the update status is the number of times since the 
+            beginning of the scan or since an 'unstage' event that the device has been triggered.
+        
+        Return Parameters
+        -----------------
+        out_est_time : list.
+            The combined est_time/std_dev pair.
+        val_dict : dict.
+            The updated version of val_dict.
+        """
+
+
+        if msg.obj is not None:
+
+            if msg.command is 'sleep':
+                return [msg.args[0], 0], val_dict
+
+            elif msg.command in (_run_start_cmds + _timed_cmds):
+                obj =  msg.obj
+                #The if-elif section below is used to track how many 'triggers' have occured since
+                #the last 'unstage'.
+                if msg.command == 'unstage':
+                    val_dict['trigger'][msg.obj.name] = 0
+                elif msg.command == 'trigger':
+                    if msg.obj.name in list(val_dict['trigger'].keys()):
```

Comment:
Removing `list` here would save from making a needless copy.

Reply:
Isn't this identical:

```
if msg.obj.name in val_dict['trigger']:```

Reply:
OK will do, for some reason I thought that this wouldn't work.